src/QMCWaveFunctions/Fermion/SlaterDetBuilder.cpp

```diff
@@ -560,7 +560,15 @@ bool SlaterDetBuilder::putDeterminant(xmlNodePtr cur, int spin_group, bool slate
     }
 #endif
   }
-  adet->set(firstIndex,lastIndex-firstIndex);
+  if( delay_rank<=0 || delay_rank>lastIndex-firstIndex )
+  {
+    APP_ABORT("SlaterDetBuilder::putDeterminant delay_rank must be positive and no larger than the electron count within a determinant!\n");
```

Comment:
The error message should print the values of delay_rank and lastIndex-firstIndex.